torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -3873,24 +3859,38 @@ def test_map_locations_multi_gpu(self):
             rpc_backend_options=options,
         )
 
-        fn = TensorPipeAgentRpcTest._gpu_add_multi_gpu
-        ret = rpc.rpc_sync(dst, fn, args=(torch.zeros(2), torch.ones(2).to(0)))
-        self.assertEqual(ret[0], torch.zeros(2) + torch.ones(2))
-        self.assertEqual(ret[1], (torch.zeros(2) - torch.ones(2)).to(0))
+        rets = rpc.rpc_sync(
+            dst,
+            TensorPipeAgentRpcTest._gpu_add_multi_gpu,
+            args=(torch.zeros(2).to(1), torch.ones(2).to(0))
+        )
+        self.assertEqual(rets[0].device, torch.device(1))
+        self.assertEqual(rets[1].device, torch.device(0))
+        self.assertEqual(rets[0], (torch.zeros(2) + torch.ones(2)).to(1))
+        self.assertEqual(rets[1], (torch.zeros(2) - torch.ones(2)).to(0))
         rpc.shutdown()
 
+    @skip_if_lt_x_gpu(2)
+    def test_device_maps_multi_gpu(self):
+        dst = worker_name((self.rank + 1) % self.world_size)
+        self._test_device_maps_multi_gpu(dst)
+
+    @skip_if_lt_x_gpu(2)
+    def test_device_maps_multi_gpu_self(self):
+        dst = worker_name(self.rank)
+        self._test_device_maps_multi_gpu(dst)
+
     @staticmethod
-    def _gpu_add_default_cpu(x, y):
-        if all([x.device.type == 'cpu', y.is_cuda, y.device.index == 1]):
-            return x.to(1) + y
+    def _gpu_add_return_to_gpu(x, y):
+        if x.device.type == 'cpu' and y.device.type == 'cpu':
+            return (x + y).to(0), (x - y).to(1), (x * y).to(2), (x / y).to(3)
         else:
             raise ValueError("Wrong device affinity")
 
-    @skip_if_lt_x_gpu(2)
-    def test_map_locations_default_cpu(self):
+    def _test_device_maps_return_to_gpu(self, dst):
         options = self.rpc_backend_options
-        dst = worker_name((self.rank + 1) % self.world_size)
-        options.set_map_location(dst, {0: 1})
+
+        options.set_device_map(dst, {0: 1})
```

Comment:
I must be missing something, but if this is the only device map we define, how can we get back tensors that are allocated on devices 1, 2 and 3?

Reply:
My bad, these tests are skipped locally as there are only two GPUs. Let me grab a server with more GPUs. hmm, I would expect CI tests cover this, let me check why this is also skipped in CI

Reply:
Looks like multigpu tests are master-only now.